packages/flutter_tools/lib/src/base/utils.dart

```diff
@@ -24,6 +24,9 @@ class BotDetector {
     return platform.environment['BOT'] != 'false'
        && (platform.environment['BOT'] == 'true'
 
+        // Non-interactive terminals are assumed to be bots.
+        || !io.stdout.hasTerminal
```

Comment:
Per Devon's comment, it seems like this should be `|| (!io.stdout.hasTerminal && platform.environment['FLUTTER_HOST'] == null)`

Reply:
I updated the code - ptal